indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java

```diff
@@ -183,19 +205,25 @@ private void processBatchesDue()
   {
     // If not leader, clear the queue and do not schedule any more rounds of processing
     if (!leaderSelector.isLeader()) {
-      log.info("Not leader anymore. Clearing [%d] batches from queue.", processingQueue.size());
-      processingQueue.clear();
+      log.info("Not leader anymore. Failing [%d] batches in queue.", processingQueue.size());
+
+      AllocateRequestBatch nextBatch = processingQueue.pollFirst();
+      while (nextBatch != null) {
+        nextBatch.markCompleted();
```

Comment:
Note: Need to make sure that this doesn't keep failing batches and meanwhile the overlord gets re-elected as the leader. Probably re-check the leader condition in the while loop.

Reply:
Done.